### PR TITLE
Implied timescales plot

### PR DESCRIPTION
**Pull request by clonker**

### CODE REVIEW

File: deeptime/plots/implied_timescales.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        assert all(callable(getattr(model, 'timescales', None)) for model in models), \
+            "all models need to have a timescales method"
+        assert all(hasattr(model, 'lagtime') for model in models), "all models need a lagtime attribute or property"
+
+        is_bayesian = hasattr(models[0], 'prior') and hasattr(models[0], 'samples')
+        lagtimes = []
+        its = []
+        its_stats = [] if is_bayesian else None
+
+        for model in models:
```
Comment (thempel) [inline deeptime/plots/implied_timescales.py:111]: One could check for missing processes and pad arrays here.

Comment (clonker) [reply deeptime/plots/implied_timescales.py:171]: added padding in the ctor call

File: deeptime/plots/implied_timescales.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+            Sampled timescales.
+
+        See Also
+        --------
+        plot_implied_timescales
+        """
+        self._lagtimes = np.asarray(lagtimes, dtype=int)
+        self._its = np.asarray(its)
+        assert self.its.ndim == 2 and self.its.shape[0] == self.n_lagtimes, \
+            "its should be of shape (lagtimes, processes)."
```
[the remark this answers is not in the document]
Comment (clonker) [reply deeptime/plots/implied_timescales.py:37]: its is now assumed to be a list of timescales per lagtime and this check only looks for the number of lagtimes

File: deeptime/plots/implied_timescales.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+    @property
+    def n_lagtimes(self) -> int:
+        r""" Number of lagtimes. """
+        return len(self.lagtimes)
+
+    @property
+    def n_processes(self) -> int:
+        r""" Number of processes. """
+        return self.its.shape[1]
```
[the remark this answers is not in the document]
Comment (clonker) [reply deeptime/plots/implied_timescales.py]: done

**Pull request merged**